static/src/javascripts/projects/common/modules/article/spacefinder.js

```diff
@@ -147,12 +155,14 @@ define([
         if (rules.selectors) {
             forOwn(rules.selectors, function (params, selector) {
                 filtered = filtered.then(function (slots) {
+                    return [slots, qwery(rules.bodySelector + selector)];
+                }).then(function (args) {
```

Comment:
TBH, I didn't change that much of the logic in the spacefinder, but indeed I find it quite difficult to follow. Something to think about indeed!